pkg/sse/sse.go

```diff
@@ -19,6 +21,18 @@ type Writer struct {
 	mtx sync.Mutex
 }
 
+func (w *Writer) writeInterface(data interface{}) error {
+	w.mtx.Lock()
+	defer w.mtx.Unlock()
+
+	wr := hh.FlushWriter{Writer: NewWriter(w.w), Enabled: true}
+	enc := json.NewEncoder(wr)
```

Comment:
I think all of this should be moved to `sse/stream.go`, so that we don't need to recreate the encoder on every write.